utils/index.d.ts

```diff
@@ -1729,14 +1737,31 @@ export declare interface ContextValueFilter {
     exclude?: string | RegExp | (string | RegExp)[];
 }
 
+interface QuickPickOptions {
+    readonly contextValues: Array<string>;
+    readonly isLeaf: boolean;
+}
+
+type CreateCallback<TNode = unknown> = (context: IActionContext) => TNode | Promise<TNode>;
+
+type CreateOptions<TNode = unknown> = {
+    label?: string;
+    callback: CreateCallback<TNode>;
+}
+
+interface CompatibleQuickPickOptions extends QuickPickOptions {
+    readonly createChild?: CreateOptions;
+}
+
 export declare interface ContextValueFilterableTreeNodeV2 {
-    readonly quickPickOptions: {
-        readonly contextValues: Array<string>;
-        readonly isLeaf: boolean;
-    }
+    readonly quickPickOptions: QuickPickOptions;
+}
+
+export declare interface CompatibleContextValueFilterableTreeNode {
+    readonly quickPickOptions: CompatibleQuickPickOptions;
 }
 
-export declare type ContextValueFilterableTreeNode = ContextValueFilterableTreeNodeV2 | AzExtTreeItem;
+export declare type ContextValueFilterableTreeNode = ContextValueFilterableTreeNodeV2;
```

Comment:
Seems like we may as well get rid of the `V2` copy of the interface and keep only `ContextValueFilterableTreeNode`.

Reply:
Additionally that probably means changing a few things in `FindByIdQuickPickStep.ts`.

Reply:
Yeah I left it to get feedback before removing it all.

To clarify, the nodes here should never be an `AzExtTreeItem` since they will be wrapped by a `CompatibleBranchDataItem` in Resource Groups.